### ISSUE TITLE
Not binary compatible with Grails 3.1.x

### ISSUE THREAD

**Issue by dariusan** [grails-plugins/grails-spring-security-acl#22]
Cannot use this plugin with Grails 3.1.x -> `Caused by: java.lang.ClassNotFoundException: org.grails.datastore.gorm.GormEntity$Trait$FieldHelper`

According to: https://grails.org/single-page-documentation.html  chapter: 3.1 
there is an issue:

> You are using a plugin or class that was compiled with a previous version of GORM and these will need to be recompiled to be Grails 3.1 and GORM 5 compatible.

**Comment by ibirrer** [comment]
Note that I also had to replace all occurrences of 

    import org.springframework.transaction.annotation.Transactional

with 

    import grails.transaction.Transactional

to make this plugin work in a grails 3.1.5 app with multiple data sources.

**Comment by dariusan** [comment]
Fix to avoid "deleted object would be resaved by cascade" exceptions while reading ACLs and deleting them within the same Hibernate session:

edit class AclService.deleteEntries() accordingly:

```
//replace the following line
AclEntry.where { aclObjectIdentity == oid }.deleteAll()
//with 
AclEntry.findAllByAclObjectIdentity(oid)*.delete()
//which does not use DML statements and updates first level cache correctly

```

**Comment by heikorm** [comment]
i followed your instructions, changed the import path of @Transactional and replaced the line in the AclService.deleteEntries()-method. but i still get an error: 

"Execution failed for task ':compileGroovy'.
> org/grails/datastore/gorm/GormEntity$Trait$FieldHelper"

my securities-plugins out of build gradle:
compile 'org.grails.plugins:spring-security-core:3.1.1'
compile 'org.grails.plugins:spring-security-ui:3.0.0.M2'
compile 'org.grails.plugins:spring-security-acl:3.0.1'

my grailsVersion=3.1.12

can anyone help me please?
thanks in advance

**Comment by jameskleeh** [comment]
Fixed in 3.1.0
